adserver/models.py

```diff
@@ -782,6 +783,33 @@ class Flight(TimeStampedModel, IndestructibleModel):
         default=0, help_text=_("Clicks across all ads in this flight")
     )
 
+    # We store nightly the top 20 publishers/countries/regions for each flight
+    # and the percentage they have filled this flight
+    # eg.
+    # {
+    #    "publishers": {"publisher1": 0.1, "publisher2": 0.05},
+    #    "countries": {"us": 0.1, "ca": 0.05, "de": 0.05},
+    #    "regions": {"us-ca": 0.25, "eu": 0.5},
+    #  }
+    traffic_fill = JSONField(
+        _("Traffic fill"),
+        blank=True,
+        null=True,
+        default=None,
+        validators=[TrafficFillValidator()],
+    )
+
+    # If set, any publisher, country, or region whose `traffic_fill` exceeds the cap
+    # will not be eligible to show on this campaign until they're below the cap.
+    # Format is the same as `traffic_fill` but this is set manually
+    traffic_cap = JSONField(
+        _("Traffic cap"),
```

Comment:
Do we need a link to docs or similar around the format here in the help_text? I hit this with the JSON targeting dict, where I don't know offhand what values are supported.